File: baekjoon_all/28000+/boj_28940.py
# Solve 2023-12-23

w, h = map(int, input().split())
n, a, b = map(int, input().split())

p = (w // a) * (h // b)
ans = -1

if p > 0:
    ans = (n - 1) // p + 1

print(ans)

File: baekjoon_all/28000+/boj_28940.cpp
// Solve 2023-12-23

#include <iostream>
using namespace std;

int main() {
    int w, h, n, a, b;
    cin >> w >> h >> n >> a >> b;

    int p = (w / a) * (h / b);
    int ans;

    if (p == 0) {
        ans = -1;
    }
    else {
        ans = (n - 1) / p + 1;
    }

    cout << ans << '\n';

    return 0;
}
